.github/workflows/test.yaml

```diff
@@ -0,0 +1,76 @@
+name: Test Examples
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test-tcp-echo:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1
+      - name: Setup
+        working-directory: tcp.echo
+        run: ./setup.sh
+      - name: Execute Test
+        working-directory: tcp.echo
+        run: |
+          set -o pipefail
+          ./test.sh | tee $GITHUB_STEP_SUMMARY
+      - name: Teardown
+        if: always()
+        working-directory: tcp.echo
+        run: ./teardown.sh
+
+  test-tls-echo:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1
+      - name: Setup
+        working-directory: tls.echo
+        run: ./setup.sh
+      - name: Execute Test
+        working-directory: tls.echo
+        run: |
+          set -o pipefail
+          ./test.sh | tee $GITHUB_STEP_SUMMARY
+      - name: Teardown
+        if: always()
+        working-directory: tls.echo
+        run: ./teardown.sh
+
+  test-http-kafka-sync:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install kcat
+        working-directory: /tmp
+        run: |
+          set -x
+          sudo apt-get update
+          sudo apt-get install librdkafka-dev libyajl-dev
+          curl -L -o kcat https://github.com/attilakreiner/kcat/releases/download/1.7.1/kcat-linux-$(arch)
+          chmod +x kcat
+          sudo mv kcat /usr/local/bin
+          kcat 2>&1 | grep "Version 1.7.1"
```

Comment:
FYI - not required, but something like this may be helpful for caching.

https://stackoverflow.com/a/59277514
https://github.com/marketplace/actions/cache-apt-packages
https://github.com/marketplace/actions/cache-anything-new

Reply:
Thx for pointing out. I added `cache-apt-pkgs-action`, LGTM.